File: android-reactive-location/src/main/java/pl/charmas/android/reactivelocation2/observables/ObservableContext.java
package pl.charmas.android.reactivelocation2.observables;

import android.content.Context;
import android.os.Handler;

import pl.charmas.android.reactivelocation2.ReactiveLocationProviderConfiguration;

public class ObservableContext {
    private final Context context;
    private final Handler handler;
    private final boolean retryOnConnectionSuspended;

    public ObservableContext(Context context, ReactiveLocationProviderConfiguration configuration) {
        this.context = context;
        this.handler = configuration.getCustomCallbackHandler();
        this.retryOnConnectionSuspended = configuration.isRetryOnConnectionSuspended();
    }

    public Context getContext() {
        return context;
    }

    Handler getHandler() {
        return handler;
    }

    boolean isRetryOnConnectionSuspended() {
        return retryOnConnectionSuspended;
    }
}

File: android-reactive-location/src/main/java/pl/charmas/android/reactivelocation2/observables/ObservableContext.kt
package pl.charmas.android.reactivelocation2.observables

import android.content.Context
import android.os.Handler
import pl.charmas.android.reactivelocation2.ReactiveLocationProviderConfiguration

class ObservableContext constructor(
    val context: Context,
    configuration: ReactiveLocationProviderConfiguration
) {
    val handler: Handler? = configuration.customCallbackHandler
    val isRetryOnConnectionSuspended: Boolean = configuration.isRetryOnConnectionSuspended
}